packages/plugin-core/src/utils/analytics.ts

```diff
@@ -133,6 +137,91 @@ export class AnalyticsUtils {
     );
   }
 
+  /** Saves analytics to be sent during the next run of Dendron.
+   *
+   * Make sure any properties you use can be trivially serialized and
+   * deserialized, numbers, strings, plain JSON objects, arrays are fine. No
+   * Maps or complex objects.
+   *
+   * This is required for actions that reload the window, where the analytics
+   * won't get sent in time before the reload and where delaying the reload
+   * would be undesirable.
+   */
+  static async trackForNextRun(event: string, customProps?: any) {
+    const ctx = "AnalyticsUtils.trackForNextRun";
+    Logger.debug({
+      ctx,
+      event,
+    });
+    const analyticsProps = this._trackCommon({
+      event,
+      props: {
+        ...customProps,
+        savedAnalytics: true,
+      },
+      timestamp: new Date(),
+    });
+    const telemetryDir = path.join(
+      os.homedir(),
+      FOLDERS.DOTFILES,
+      FOLDERS.SAVED_TELEMETRY
+    );
+    await fs.ensureDir(telemetryDir);
+    await fs.writeFile(
+      path.join(telemetryDir, `${genUUID()}.json`),
+      JSON.stringify(analyticsProps)
+    );
+  }
+
+  static async sendSavedAnalytics() {
+    const ctx = "AnalyticsUtils.sendSavedAnalytics";
+    Logger.debug({ ctx, message: "start" });
+    const telemetryDir = path.join(
+      os.homedir(),
+      FOLDERS.DOTFILES,
+      FOLDERS.SAVED_TELEMETRY
+    );
+    let files: string[] = [];
+    try {
+      files = await fs.readdir(telemetryDir);
+    } catch {
+      Logger.warn({
+        ctx,
+        msg: "failed to read the saved telemetry dir",
+        telemetryDir,
+      });
+    }
+
+    return asyncLoop(
+      files.filter((filename) => path.extname(filename) === ".json"),
+      async (filename) => {
+        const filePath = path.join(telemetryDir, filename);
+        try {
+          const contents = await fs.readFile(filePath, { encoding: "utf-8" });
+          const payload = JSON.parse(contents, (_key, value) => {
+            // If we have a timestamp, that will get serialized as a string. We have to deserialize it here.
+            if (
+              _.isString(value) &&
+              // yyyy-mm-ddThh:mm:ss.msZ
+              value.match(/\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d[.]\d+Z/)
+            ) {
+              return new Date(value);
+            }
+            return value;
+          });
+          await fs.rm(filePath);
+          return SegmentUtils.trackSync(payload);
+        } catch (err) {
+          Logger.warn({
```

Comment:
The most likely reason why these would fail is file permission issues, which we can't really do anything about. Since we have too much stuff in sentry anyway I erred on the side of not tracking it, would you suggest that we do track?

Reply:
fair point. lets keep it as is given the state of sentry